lib/jasmine-reportportal-reporter.js

```diff
@@ -239,29 +241,39 @@ class ReportportalReporter {
     }
 
     specStarted(spec) {
-        this.startTime = this.getTime() - 1;
         const fullTestName = SpecificUtils.getFullTestName(spec);
         const promise = SpecificUtils.getCodeRef(this.currentTestFilePathIndex, fullTestName);
-
         return promise.then((codeRef) => {
-            let stepObj = this.client.startTestItem(Object.assign({
+            this.startTime = this.getTime();
+            const parent = this.getParentInfo();
+            const stepObj = this.client.startTestItem(Object.assign({
                 type: entityType.STEP,
                 description: spec.description,
+                startTime: this.startTime,
                 name: spec.description
-            }, codeRef && { codeRef }), this.tempLaunchId, this.getParentId());
+            }, codeRef && { codeRef }), this.tempLaunchId, parent && parent.tempId);
 
-            this.setParentId(stepObj.tempId);
+            this.setParentInfo({ tempId: stepObj.tempId, startTime: this.startTime });
             promiseErrorHandler(stepObj.promise);
         });
     }
 
+    getHookStartTime(hookType, parent) {
+        if (hookType === entityType.BEFORE_METHOD || hookType === entityType.BEFORE_SUITE) {
+            return Math.max(parent && parent.startTime, this.startTime - 1);
```

Comment:
Why `max`? Should we use the earliest time to report `before` hooks?

Reply:
This is all due to the specific work of hooks and at the same time, the speed of the tests themselves. We need max as the value of `parent.startTime` often less than `this.startTime - 1`. The value of parentStartTime is not that we need. But it helps us avoid problems if there is no `this.startTime`.